### DISCUSSION TITLE
Not all compose.yml dependencies are being detected (playwright issue)

### DISCUSSION THREAD

**Discussion by Sergih28** [renovatebot/renovate#28486] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I am extending the `monorepo:playwright` preset in my config.

:heavy_check_mark: My playwright dependency (@playwright/test) from `package.json` is being detected.
:heavy_check_mark: My playwright dependency (@playwright/test) from my `Dockerfile` via regex is being detected.
:x: My playwright dependency (mcr.microsoft.com/playwright:v1.42.1-jammy) from my `compose.yaml` is not being detected.

The line not detected is:
`image: mcr.microsoft.com/playwright:v1.42.1-jammy`

Even if I pin the sha, it still does not detect it:
`image: mcr.microsoft.com/playwright:v1.42.1-jammy@sha256:b2b78a74b04b3cfdf35046c623c49741de35ef67f1b4a21de8e316dbb89fe68e`

Isn't the purpose of the playwright preset to detect this?

I can tell renovate is reading my `compose.yaml` because it is detecting correctly without any extra config my pgadmin and postgres images.

I've tried to detect that line via regex playing around with `depNameTemplate` and `registryUrlTemplate` (which I'm not even sure if [this](https://mcr.microsoft.com/en-us/product/playwright/tags) is the correct one for playwright), and I kind of managed it, but the version update never pops up.

I tried a lot of things but I don't know what should I do anymore.

That said, that would be my last resort, I think it should work automatically with the playwright preset, no?

I'll provide any necessary logs.

Thanks for helping! :smiley: 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by Sergih28** [comment, marked as answer]
Alright guys, I managed to do it, so I'm going to share my solution. :tada: 
But I think this should already be done automatically when extending `monorepo:playwright`.

This detects properly the dependency:

```
"customManagers": [
  {
      "customType": "regex",
      "fileMatch": [ "^compose.yaml$" ],
      "matchStrings": [ "image: mcr.microsoft.com/playwright:(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)\\s" ],
      "depNameTemplate": "mcr.microsoft.com/playwright",
      "datasourceTemplate": "docker"
  }
]
```

This makes it show in the same pr with the other playwright dependencies in my Dockerfile and package.json.

```
"packageRules": [
  {
      "groupName": "Playwright monorepo",
      "groupSlug": "playwright-monorepo",
      "matchPackageNames": [
        "@playwright/test",
        "mcr.microsoft.com/playwright"
      ]
  }
]
```

EDIT:
I had to update the packageRules to filter out the digest, otherwise Docker complains and if you don't filter it out in the config here, renovate will error out or try to update it again.

So my final changes are:
```
"customManagers": [
  {
      "customType": "regex",
      "fileMatch": [ "^compose.yaml$" ],
      "matchStrings": [ "image: mcr.microsoft.com/playwright:(?<currentValue>.*?)\\s" ],
      "depNameTemplate": "mcr.microsoft.com/playwright",
      "datasourceTemplate": "docker"
  }
]
```

```
"packageRules": [
  {
      "groupName": "Playwright monorepo",
      "groupSlug": "playwright-monorepo",
      "matchPackageNames": [
        "@playwright/test",
        "mcr.microsoft.com/playwright"
      ],
      "extends": [ ":pinDigestsDisabled" ]
  }
```

**Reply by viceice** [reply]
match the digest as `currentDigest`, so renovate can also update it

**Reply by Sergih28** [reply]
Yep, so just to clarify to whoever is reading this in the future, my first way worked, and renovate updated the digest correctly.

It was just docker not liking the digest pinning for playwright for some reason, and not complaining with my other pinned digests in my compose.yaml (pgadmin and postgres) :shrug:. That's why I did those changes, but really had nothing to do with renovate at that point.

Just thought I'd leave that written as well in case someone else happens to use playwright's image in their compose.yaml :+1:
